### DISCUSSION TITLE
Embed videos from webpages into tiddlers using tiddlyclip as local media.

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#64] [Ideas]
https://twitter.com/SocAbdRadiology/status/1410765649488625669

Is there any way to embed videos from webpages (like in the link given above) just like the pin/snap feature for images and save them as local media files.

**Comment by buggyj** [comment]
In theory yes, but things are much more complicated with video, and less standard, so it is less likely that a good and reliable solution could be constructed.

**Reply by arunnbabu81** [reply]
I intend to use it for videos from twitter only. Is there anything that can be done for that ?

**Reply by arunnbabu81** [reply]
https://twittervideodownloader.com/download

From what i understand from this site, twitter videos are available for download in mp4 format. Can a video clipping rule be made for mp4 videos?

**Reply by buggyj** [reply]
It is almost the same as the rest

![image](https://user-images.githubusercontent.com/2855795/151406145-ffd9d028-f602-4c50-a02b-c8f435cc8e49.png)

**Reply by arunnbabu81** [reply]
> It is almost the same as the rest
> 
> ![image](https://user-images.githubusercontent.com/2855795/151406145-ffd9d028-f602-4c50-a02b-c8f435cc8e49.png)

This is for local mp4 files rite? I was thinking of something which could clip the mp4 video directly from twitter and save it as local media .

**Reply by buggyj** [reply]
Yes you need to use the site you listed first to download it.

**Reply by arunnbabu81** [reply]
How does the extvideo rule in this [mediaplayer ](http://mediaplayer.tiddlyspot.com/)wiki works 

![image](https://user-images.githubusercontent.com/67494083/151424715-c0d2e357-abc4-4e54-b221-2e59ef016513.png)

**Reply by buggyj** [reply]
I am not sure what you are asking, it will make a canonical_uri tiddler of type video/mp4 which will contain the address of the video on the local harddisk 

I wrote this 7 years ago - it looks like it will not work.
